File: app/src/main/java/com/artemkaxboy/android/autoredialce/utils/Logger.java
package com.artemkaxboy.android.autoredialce.utils;

import android.util.Log;

@SuppressWarnings("unused")
public class Logger {

  public static void wtf(String msg) {
    Log.wtf(Cons.TAG, msg);
  }

  public static void wtf(Throwable e) {
    Log.wtf(Cons.TAG, e);
  }

  public static void wtf(String msg, Throwable e) {
    Log.wtf(Cons.TAG, msg, e);
  }

  public static void error(String msg) {
    Log.e(Cons.TAG, msg);
  }

  public static void error(String msg, Throwable e) {
    Log.e(Cons.TAG, msg, e);
  }

  public static void warning(String msg) {
    Log.w(Cons.TAG, msg);
  }

  public static void warning(Throwable e) {
    Log.w(Cons.TAG, e);
  }

  public static void warning(String msg, Throwable e) {
    Log.w(Cons.TAG, msg, e);
  }

  public static void info(String msg) {
    Log.i(Cons.TAG, msg);
  }

  public static void info(String msg, Throwable e) {
    Log.i(Cons.TAG, msg, e);
  }

  public static void verbose(String msg) {
    Log.v(Cons.TAG, msg);
  }

  public static void verbose(String msg, Throwable e) {
    Log.v(Cons.TAG, msg, e);
  }

  public static void debug(String msg) {
    Log.d(Cons.TAG, msg);
  }

  public static void debug(String msg, Throwable e) {
    Log.d(Cons.TAG, msg, e);
  }
}

File: app/src/main/java/com/artemkaxboy/android/autoredialce/utils/Logger.kt
package com.artemkaxboy.android.autoredialce.utils

import android.util.Log
import com.artemkaxboy.android.autoredialce.BuildConfig

object Logger {

    @Suppress("unused") /* for public usage */
    fun wtf(msg: () -> String) {
        Log.wtf(TAG, msg.invoke())
    }

    @Suppress("unused") /* for public usage */
    fun wtf(e: Throwable) {
        Log.wtf(TAG, e)
    }

    @Suppress("unused") /* for public usage */
    fun wtf(msg: () -> String, e: Throwable) {
        Log.wtf(TAG, msg.invoke(), e)
    }

    @Suppress("unused") /* for public usage */
    fun error(msg: () -> String) {
        Log.e(TAG, msg.invoke())
    }

    @Suppress("unused") /* for public usage */
    fun error(msg: () -> String, e: Throwable) {
        Log.e(TAG, msg.invoke(), e)
    }

    @Suppress("unused") /* for public usage */
    fun warning(msg: () -> String) {
        Log.w(TAG, msg.invoke())
    }

    @Suppress("unused") /* for public usage */
    fun warning(e: Throwable) {
        Log.w(TAG, e)
    }

    @Suppress("unused") /* for public usage */
    fun warning(msg: () -> String, e: Throwable) {
        Log.w(TAG, msg.invoke(), e)
    }

    @Suppress("unused") /* for public usage */
    fun info(msg: () -> String) {
        Log.i(TAG, msg.invoke())
    }

    @Suppress("unused") /* for public usage */
    fun info(msg: () -> String, e: Throwable) {
        Log.i(TAG, msg.invoke(), e)
    }

    @Suppress("unused") /* for public usage */
    fun debug(tag: () -> String, msg: () -> String) {
        if (BuildConfig.DEBUG) {
            Log.d(tag.invoke(), msg.invoke())
        }
    }

    @Suppress("unused") /* for public usage */
    fun debug(msg: () -> String) {
        if (BuildConfig.DEBUG) {
            Log.d(TAG, msg.invoke())
        }
    }

    @Suppress("unused") /* for public usage */
    fun debug(msg: () -> String, e: Throwable) {
        if (BuildConfig.DEBUG) {
            Log.d(TAG, msg.invoke(), e)
        }
    }

    @Suppress("unused") /* for public usage */
    fun verbose(msg: () -> String) {
        Log.v(TAG, msg.invoke())
    }

    @Suppress("unused") /* for public usage */
    fun verbose(msg: () -> String, e: Throwable) {
        Log.v(TAG, msg.invoke(), e)
    }
}
